sdk/src/main/java/io/opentdf/platform/sdk/Autoconfigure.java

```diff
@@ -0,0 +1,749 @@
+package io.opentdf.platform.sdk;
+
+import java.io.IOException;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.StringJoiner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Supplier;
+
+import io.opentdf.platform.policy.KeyAccessServer;
+import io.opentdf.platform.policy.attributes.GetAttributeValuesByFqnsRequest;
+import io.opentdf.platform.policy.attributes.GetAttributeValuesByFqnsResponse;
+import io.opentdf.platform.policy.attributes.GetAttributeValuesByFqnsResponse.AttributeAndValue;
+import io.opentdf.platform.policy.Attribute;
+import io.opentdf.platform.policy.Value;
+import io.opentdf.platform.policy.AttributeValueSelector;
+import io.opentdf.platform.policy.AttributeRuleTypeEnum;
+
+// Error handling class
+class AutoConfigureException extends IOException {
+    public AutoConfigureException(String message) {
```

Comment:
I recommend adding a constructor with a `cause` parameter, also, since that helps with debugging